GH-8586: Deprecate IntegrationComponentSpec.get() (#8594)
spring-projects/spring-integration: src/reference/asciidoc/whats-new.adoc

 - The default timeout for send and receive operations in gateways and replying channel adapters has been changed from infinity to `30` seconds.
Only one left as a `1` second is a `receiveTimeout` for `PollingConsumer` to not block a scheduler thread too long and let other queued tasks to be performed with the `TaskScheduler`.

>  - An `IntegrationComponentSpec.get()` method has been deprecated with removal plans in the next version.

Reviewer: ```suggestion
 - The `IntegrationComponentSpec.get()` method has been deprecated with removal planned for the next version.
```

Since `IntegrationComponentSpec` is a `FactoryBean`, its bean definition must stay as is without any target object resolutions.

> The Java DSL and the framework by itself will take care about the proper `IntegrationComponentSpec` lifecycle.

Reviewer: ```suggestion
The Java DSL and the framework by itself will manage the `IntegrationComponentSpec` lifecycle.
```
